tests/unit/reportGenService.unitTest.ts

```diff
@@ -24,128 +24,129 @@ describe("ReportGenerationService", () => {
-                      }
-                    };
-                  });
-                  const mockActivitiesService = jest.fn().mockImplementation(() => {
-                    return {
-                      getActivities: () => {
-                        return Promise.resolve(JSON.parse(waitResponse.body));
-                      }
-                    };
-                  });
-                  const reportGenSvc = new ReportGenerationService(new mockTestResultsService(), new mockActivitiesService());
-
-                  const output = await reportGenSvc.generateATFReport(activity);
-                  const workbook = new Excel.Workbook();
-                  const stream = new Duplex();
-                  stream.push(output.fileBuffer); // Convert the incoming file to a readable stream
-                  stream.push(null);
-
-                  const excelFile = await workbook.xlsx.read(stream);
-                  const reportSheet: Excel.Worksheet = excelFile.getWorksheet(1);
-
-                  expect(reportSheet.getCell("H25").value).toEqual(2);
-                  expect(reportSheet.getCell("H26").value).toEqual(5);
-                  expect(reportSheet.getCell("F25").value).toEqual("JY58FPP");
-                  expect(reportSheet.getCell("E26").value).toEqual("10:36:33");
-                });
-            });
+    context("and testResults returns HGVs and TRLs", () => {
+      it("should return a valid xlsx file as buffer with trailerId populated for trl vehicles and noOfAxles populated for hgv and trl vehicles", async () => {
+        const mockTestResultsService = jest.fn().mockImplementation(() => {
+          return {
+            getTestResults: () => {
+              return Promise.resolve(TestResultsService.prototype.expandTestResults(JSON.parse(hgvTrlTestResultResponse.body)));
+            }
+          };
         });
+        const mockActivitiesService = jest.fn().mockImplementation(() => {
+          return {
+            getActivities: () => {
+              return Promise.resolve(JSON.parse(waitResponse.body));
+            }
+          };
+        });
+        const reportGenSvc = new ReportGenerationService(new mockTestResultsService(), new mockActivitiesService());
+
+        const output = await reportGenSvc.generateATFReport(activity);
+        const workbook = new Excel.Workbook();
+        const stream = new Duplex();
+        stream.push(output.fileBuffer); // Convert the incoming file to a readable stream
+        stream.push(null);
+
+        const excelFile = await workbook.xlsx.read(stream);
+        const reportSheet: Excel.Worksheet = excelFile.getWorksheet(1);
+
+        expect(reportSheet.getCell("H25").value).toEqual(2);
+        expect(reportSheet.getCell("H26").value).toEqual(5);
+        expect(reportSheet.getCell("F25").value).toEqual("JY58FPP");
+        expect(reportSheet.getCell("E26").value).toEqual("10:36:33");
+        expect(reportSheet.getCell("F26").value).toEqual("12345");
```

Comment:
This is surprising. Did you change the actual report generation piece?

Reply:
No but the E26 != 12345 (which actually was F26) kept cropping up when working on this that I had to ensure that would never cause issues again